### ISSUE TITLE
[Suggestion] Hooks to modify door behavior

### ISSUE THREAD

**Issue by Risky-Shot** [overextended/ox_doorlock#202]
Similar to ox_inventory, we can use hooks to cancel the door open/close (or any other thing).

Use Case : A business is under lockdown, we can implement logic so that worker/owner cannot open the door of business MLO whereas certain groups can (say police). 

This will allow data of authorization to be preserved and we don't have to edit a door for such thing to take place.

**Comment by Skryptific** [comment]
I agree, I think it would be nice to have some control before a door state gets changed.

@thelindat, is this something you would like to see implemented?
